test/integration/provider/azure.go

```diff
@@ -0,0 +1,182 @@
+package provider
+
+/**
+	Orphaned Resources
+	- VMs:
+		Describe instances with specified tag name:<cluster-name>
+		Report/Print out instances found
+		Describe volumes attached to the instance (using instance id)
+		Report/Print out volumes found
+		Delete attached volumes found
+		Terminate instances found
+	- Disks:
+		Describe volumes with tag status:available
+		Report/Print out volumes found
+		Delete identified volumes
+**/
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/utils/pointer"
+
+	provider "github.com/gardener/machine-controller-manager-provider-azure/pkg/azure"
+	"github.com/gardener/machine-controller-manager-provider-azure/pkg/spi"
+	v1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+	"github.com/gardener/machine-controller-manager/pkg/util/provider/driver"
+)
+
+// AdditionalResourcesCheck check for orphan network interfaces and triggers their deletion
+func additionalResourcesCheck(
+	clients spi.AzureDriverClientsInterface,
+	resourceGroup,
+	tagName,
+	tagValue string,
+) error {
+	ctx := context.TODO()
+
+	networkInterfaces, err := clients.GetNic().List(ctx, resourceGroup)
+	if err != nil {
+		return err
+	}
+	for _, networkInterface := range networkInterfaces.Values() {
+		if value, ok := networkInterface.Tags[tagName]; ok && tagValue == *value {
+			fmt.Println(networkInterface.Name)
+			nicDeleteFuture, err := clients.GetNic().Delete(ctx, resourceGroup, *networkInterface.Name)
+			if err != nil {
+				return err
+			}
+			if err = nicDeleteFuture.WaitForCompletionRef(ctx, clients.GetClient()); err != nil {
+				return err
+			}
+		}
+	}
+	return err
+}
+
+// deleteVolume deletes the specified volume
+func deleteVolume(
+	clients spi.AzureDriverClientsInterface,
```

Comment:
we should print (like how we print when we get error while deleting an instance) that whats the volumeID of the instance which was not able to delete.
This will tell user to go and delete the instance on the cloud provider by himself

Similarly for NICs in `additionalResourcesCheck()`